docs/designs/serving_design.md

```diff
@@ -0,0 +1,103 @@
+# Serving Design
+
+## Background
+
+Model serving is an essential part in an end-to-end machine learning lifecycle. Publishing the trained model as a service in production can make it valuable in the real world. In this article, we focus on how to save the model trained in ElasticDL for serving.
+
+[SavedModel](https://www.tensorflow.org/guide/saved_model) is the universal file format for TensorFlow models. It's language neutral and can be loaded by multiple frameworks (such as TFServing, TFLite, TensorFlow.js and so on). We choose to export the ElasticDL model to the SavedModel format. In this way, we can leverage various mature solutions to serving our model in different scenarios.
+
+The model size can vary from several kilobytes to several terabytes. Exporting to SavedModel can't apply to all the scenarios. We divide the model size into two categories: *Small or medium size* and *large size*. The small or medium size model can be loaded by a process, but the latter cannot fit in a single process. Training and serving strategies are different between these two cases. The following table describes the choices of serving strategy considering model size and several training strategies:
+
+|                            | Master Central Storage |  AllReduce  |            Parameter Server              |
+|:--------------------------:|:----------------------:|:-----------:|:----------------------------------------:|
+| Small or Medium Size Model |       SavedModel       |  SavedModel |               SavedModel                 |
+| Large Size Model           |          N/A           |     N/A     | Distributed Parameter Server for Serving |
+
+We consider three main training strategies of ElasticDL. Master central storage strategy (Implemented): The entire model is loaded in the master node. Workers pull the model from the master while processing each minibatch of data. AllReduce strategy (Planning): It mainly support sync training. Each worker loads a replica of the entire model. AllReduce aggregates gradients across all the workers and publish them to each worker. Parameter server strategy (In Progress): We store the model variables on the parameter servers. All the workers pull the variables as needed from the parameter server and execute computation.
```

Comment:
Better to add (1) (2) (3) before the 3 strategies to make it look clearer.

Reply:
support -> supports

Reply:
Good idea, updated.